base/redactor.go

```diff
@@ -17,13 +17,18 @@ type Redactor interface {
 }
 
 type RedactorSlice []Redactor
+type RedactorFuncSlice []RedactorFunc
+
+type RedactorFunc func() Redactor
 
 // redact performs an *in-place* redaction on the input slice, and returns it.
 // This should only be consumed by logging funcs. E.g. fmt.Printf(fmt, redact(args))
 func redact(args []interface{}) []interface{} {
 	for i, v := range args {
 		if r, ok := v.(Redactor); ok {
 			args[i] = r.Redact()
+		} else if r, ok := v.(RedactorFunc); ok {
+			args[i] = r().Redact()
```

Comment:
Again, implementing the `Redactor` interface on `RedactorFunc` removes the need for this concrete type check.